MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1371,6 +1412,132 @@ void EnggDiffractionPresenter::setDataToClonedWS(std::string &current_WS,
 }
 
+void EnggDiffractionPresenter::setBankItems() {
+  try {
+    auto fitting_runno_vector = m_view->getFittingRunNumVec();
+
+    if (!fitting_runno_vector.empty()) {
+
+      // delete previous bank added to the list
+      m_view->clearFittingComboBox();
+
+      for (size_t i = 0; i < fitting_runno_vector.size(); i++) {
+        Poco::Path vecFile(fitting_runno_vector[i]);
+        std::string strVecFile = vecFile.toString();
+        // split the directory from m_fitting_runno_dir_vec
+        std::vector<std::string> vecFileSplit =
+            m_view->splitFittingDirectory(strVecFile);
+
+        // get the last split in vector which will be bank
+        std::string bankID = (vecFileSplit.back());
+
+        bool digit = isDigit(bankID);
+
+        if (digit || bankID == "cropped") {
+          m_view->addBankItem(bankID);
+        } else {
+          QString qBank = QString("Bank %1").arg(i + 1);
+          m_view->addBankItem(qBank.toStdString());
+        }
+      }
+
+      m_view->enableFittingComboBox(true);
+    } else {
+      // upon invalid file
+      // disable the widgets when only one related file found
+      m_view->enableFittingComboBox(false);
+
+      m_view->clearFittingComboBox();
+    }
+
+  } catch (std::runtime_error &re) {
+    m_view->userWarning("Unable to insert items: ",
+                        "Could not add banks to "
+                        "combo-box or list widget; " +
+                            static_cast<std::string>(re.what()) +
+                            ". Please try again");
+  }
+}
+
+void EnggDiffractionPresenter::setRunNoItems(
+    std::vector<std::string> runNumVector, bool multiRun) {
+  try {
+    if (!runNumVector.empty()) {
+
+      // delete previous bank added to the list
+      m_view->clearFittingListWidget();
+
+      for (size_t i = 0; i < runNumVector.size(); i++) {
+
+        // get the last split in vector which will be bank
+        std::string currentRun = (runNumVector[i]);
+
+        m_view->addRunNoItem(currentRun);
+      }
+
+      if (multiRun) {
+        m_view->enableFittingListWidget(true);
+
+        auto currentIndex = m_view->getFittingListWidgetCurrentRow();
+        if (currentIndex == -1)
+          m_view->setFittingListWidgetCurrentRow(0);
+      } else {
+        m_view->enableFittingListWidget(false);
+      }
+    }
+
+    else {
+      // upon invalid file
+      // disable the widgets when only one related file found
+      m_view->enableFittingListWidget(false);
+
+      m_view->clearFittingListWidget();
+    }
+
+  } catch (std::runtime_error &re) {
+    m_view->userWarning("Unable to insert items: ",
+                        "Could not add list widget; " +
+                            static_cast<std::string>(re.what()) +
+                            ". Please try again");
+  }
+}
+
+void EnggDiffractionPresenter::setDefaultBank(
+    std::vector<std::string> splittedBaseName, QString selectedFile) {
```

Comment:
`splittedBaseName` can also be a `const &`.
And also for later, `QString` should not be used here in the presenter.